sklearn/random_projection.py

```diff
@@ -59,9 +59,9 @@ def johnson_lindenstrauss_min_dim(n_samples, *, eps=0.1):
 
       (1 - eps) ||u - v||^2 < ||p(u) - p(v)||^2 < (1 + eps) ||u - v||^2
 
-    Where u and v are any rows taken from a dataset of shape [n_samples,
-    n_features], eps is in ]0, 1[ and p is a projection by a random Gaussian
-    N(0, 1) matrix with shape [n_components, n_features] (or a sparse
+    Where u and v are any rows taken from a dataset of shape (n_samples,
+    n_features), eps is in ]0, 1[ and p is a projection by a random Gaussian
```

Comment:
I would prefer:

```suggestion
    n_features), eps is in (0, 1) and p is a projection by a random Gaussian
```

for open intervals.

Reply:
Both notations are correct given the ISO 31-11 (at least from what I can see on Wikipedia :)). So I would let it as is.